hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/HoodieTimelineArchiver.java

```diff
@@ -131,7 +133,8 @@ public HoodieTimelineArchiver(HoodieWriteConfig config, HoodieTable<T, I, K, O>
           metaClient.getActiveTimeline().getCommitsTimeline(),
           cleanerPolicy,
           cleanerCommitsRetained,
-          latestCommit.isPresent()
+          latestCommit.isPresent() && (latestCommit.get().getTimestamp().length() == MILLIS_INSTANT_ID_LENGTH
+              ||  latestCommit.get().getTimestamp().length() == SECS_INSTANT_ID_LENGTH)
```

Comment:
Instead of checking here, should we fix `HoodieActiveTinline#parseDateFromInstantTime`? There could be other callers too and it's essentially a timestamp parsing issue.